core/tests/puppeteer-acceptance-tests/utilities/user/topic-manager.ts

```diff
@@ -3168,6 +3559,394 @@ export class TopicManager extends BaseUser {
+  }
+
+  /**
+   * Checks if the save changes button is enabled or disabled in the skill editor.
+   * @param {'enabled' | 'disabled'} status - The status to check.
+   */
+  async expectSaveChangesInSkillEditorToBe(
+    status: 'enabled' | 'disabled'
+  ): Promise<void> {
+    await this.expectElementToBeClickable(
+      saveOrPublishSkillSelector,
+      status === 'enabled'
+    );
+  }
+
+  /**
+   * Checks if the skill is assigned to the given topics.
+   * @param {string} skillName - The name of the skill.
+   * @param {string} topicNames - The names of the topics, separated by comma.
+   */
+  async expectSkillAssignedToTopic(
+    skillName: string,
+    topicNames: string
+  ): Promise<void> {
+    const skillDescriptionSelector = this.isViewportAtMobileWidth()
+      ? mobileSkillDescriptionSelector
+      : desktopSkillDescriptionSelector;
+    await this.expectElementToBeVisible(skillNameAndStatusContainer);
+
+    const skillContainers = await this.page.$$(skillsAssignmentSelector);
+    const skillContainer = skillContainers.find(async container => {
+      const foundSkillName = await container.$eval(
+        skillDescriptionSelector,
+        element => element.textContent?.trim()
+      );
+      return foundSkillName === skillName;
+    });
+
+    if (!skillContainer) {
+      throw new Error(`Skill ${skillName} not found in topic.`);
+    }
+
+    const skillStatus = await skillContainer.$eval(
+      skillsAssignmentSelector,
+      element => element.textContent?.trim()
+    );
+
+    expect(skillStatus).toBe(topicNames);
+  }
+
+  /**
+   * Navigates to the tab in the topic editor page.
+   * @param {'Preview Tab' | 'Questions Tab'} tabName - The name of the tab.
+   */
+  async navigateToTabInTopicEditorPage(
+    tabName: 'Preview Tab' | 'Questions Tab'
+  ): Promise<void> {
+    const tabSelector = `.e2e-test-topic-${tabName.toLocaleLowerCase().replace(' ', '-')}`;
+    await this.expectElementToBeVisible(tabSelector);
+    await this.clickOn(tabSelector);
+  }
+
+  /**
+   * Selects a skill in the questions tab.
+   * @param {string} skillName - The name of the skill to select.
+   */
+  async selectSkillInQuestionsTab(skillName: string): Promise<void> {
+    await this.expectElementToBeVisible(skillSelectInQuestionTabSelector);
+    await this.clickOn(skillSelectInQuestionTabSelector);
+
+    await this.selectMatOption(skillName);
+    await this.expectTextContentToBe(
+      skillSelectInQuestionTabSelector,
+      skillName
+    );
+  }
+
+  /**
+   * Checks if the question is visible in the questions tab.
+   * @param {string} question - The question to check.
+   */
+  async expectQuestionToBeVisible(
+    question: string
+  ): Promise<ElementHandle<Element>> {
+    await this.expectElementToBeVisible(questionTextSelector);
+
+    const questionElements = await this.page.$$(questionTextSelector);
+    let questionElement: ElementHandle<Element> | null = null;
+    for (const questionEle of questionElements) {
```

Comment:
`questionEle` is a confusing variable name when we already have `questionElement`

Reply:
`questionEle` -> `questionElement`
`questionElement` -> `requiredQuestionElement`